zeebe/atomix/utils/src/test/java/io/atomix/utils/VersionTest.java

```diff
@@ -16,63 +16,52 @@
  */
 package io.atomix.utils;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 import org.junit.Test;
 
 /** Version test. */
 public class VersionTest {
   @Test
   public void testVersionComparison() {
-    assertTrue(Version.from("1.0.0").compareTo(Version.from("2.0.0")) < 0);
-    assertTrue(Version.from("2.0.0").compareTo(Version.from("1.0.0")) > 0);
-    assertTrue(Version.from("1.0.0").compareTo(Version.from("0.1.0")) > 0);
-    assertTrue(Version.from("0.1.0").compareTo(Version.from("1.0.0")) < 0);
-    assertTrue(Version.from("0.1.0").compareTo(Version.from("0.1.1")) < 0);
-    assertTrue(Version.from("1.0.0").compareTo(Version.from("0.0.1")) > 0);
-    assertTrue(Version.from("1.1.1").compareTo(Version.from("1.0.3")) > 0);
-    assertTrue(Version.from("1.0.0").compareTo(Version.from("1.0.0-beta1")) > 0);
-    assertTrue(Version.from("1.0.0-rc2").compareTo(Version.from("1.0.0-rc1")) > 0);
-    assertTrue(Version.from("1.0.0-rc2.1").compareTo(Version.from("1.0.0-rc2")) > 0);
-    assertTrue(Version.from("1.0.0-rc2.1.1").compareTo(Version.from("1.0.0-rc2.1")) > 0);
-    assertTrue(Version.from("1.0.0-rc2").compareTo(Version.from("1.0.0-rc2.1")) < 0);
-    assertTrue(Version.from("1.0.0-rc1").compareTo(Version.from("1.0.0-beta1")) > 0);
-    assertTrue(Version.from("2.0.0-beta1").compareTo(Version.from("1.0.0")) > 0);
-    assertTrue(Version.from("1.0.0-alpha1").compareTo(Version.from("1.0.0-SNAPSHOT")) > 0);
+    assertThat(Version.from("1.0.0")).isLessThan(Version.from("2.0.0"));
+    assertThat(Version.from("2.0.0")).isGreaterThan(Version.from("1.0.0"));
+    assertThat(Version.from("1.0.0")).isGreaterThan(Version.from("0.1.0"));
+    assertThat(Version.from("0.1.0"))
+        .isLessThan(Version.from("1.0.0"))
+        .isLessThan(Version.from("0.1.1"));
+    assertThat(Version.from("1.0.0")).isGreaterThan(Version.from("0.0.1"));
+    assertThat(Version.from("1.1.1")).isGreaterThan(Version.from("1.0.3"));
+    assertThat(Version.from("1.0.0")).isGreaterThan(Version.from("1.0.0-beta1"));
+    assertThat(Version.from("1.0.0-rc2")).isGreaterThan(Version.from("1.0.0-rc1"));
+    assertThat(Version.from("1.0.0-rc2.1")).isGreaterThan(Version.from("1.0.0-rc2"));
+    assertThat(Version.from("1.0.0-rc2.1.1")).isGreaterThan(Version.from("1.0.0-rc2.1"));
+    assertThat(Version.from("1.0.0-rc2")).isLessThan(Version.from("1.0.0-rc2.1"));
+    assertThat(Version.from("1.0.0-rc1")).isGreaterThan(Version.from("1.0.0-beta1"));
+    assertThat(Version.from("2.0.0-beta1")).isGreaterThan(Version.from("1.0.0"));
+    assertThat(Version.from("1.0.0-alpha1")).isGreaterThan(Version.from("1.0.0-SNAPSHOT"));
```

Comment:
:exclamation: This is counter intuitive for us, but the semantic version spec doesn't really treat snapshot differently. So right now, I would expect 1.0.0-SNAPSHOT to be the "greatest" version always, such that `1.0.0-alpha1 < 1.0.0-SNAPSHOT < 1.0.0`, but this is currently not the case (nor specified in the spec). It's a bit of a gotcha, so maybe @oleschoenburg has some opinion here - but I suspect they simply followed the spec :shrug:

Reply:
Can we have a test with the case `-alpha1-rc1`